### ISSUE TITLE
locating language service should be improved and fixed where necessary

### ISSUE THREAD

**Issue by shalom938** [digma-ai/digma-intellij-plugin#331]
in GroupListViewItemBuilder
this code sends to method findWorkspaceUrisForSpans the  insight.getCodeObjectId, this method expects that argument to be method code object id, but the actual value is span id.

```
            case SpanDurationBreakdown:{
                WorkspaceUrisHelper.findWorkspaceUrisForSpans(project, theListView, getSpanIds((SpanDurationBreakdownInsight) insight), insight.getCodeObjectId());
                break;
            }
```

findWorkspaceUrisForSpans uses this method id to find the relevant language service by calling
LanguageService.findLanguageServiceByMethodCodeObjectId(project, methodCodeObjectId)
it will not find one because its a span id. 
locating the relevant language service has a fallback that if not found by method id it uses the most dominant language that was used until now in the project. that is not accurate and may return a wrong language service.
there are use cases where we don't know which language service to use because there is no context. for example the summary view loads some data from the backend and then requires some services that need a language service.  in that case if it can provide a method id the LanguageServiceLocator can find a language service by finding the method in the project. but if it can't provide a method id LanguageServiceLocator will return the most dominant language.

but, in GroupListViewItemBuilder we can have a context because this flow started when opening a file, we just don't pass the file to the flow, but we can pass the file to this flow and find the correct language service because a file can provide its language.
I suggest to fix wherever possible to pass a kind of context , or just the PSI file, so wherever we need a language service we don't have to guess.

its not many places , currently its only in  InsightsViewBuilder that we don't have a context to tell us which language service to use.

In summary view there is probably nothing to do and we must rely on method id and search for the language.

 the example above of SpanDurationBreakdown is a bug, it will work if at least one access to a language service was already made because document service keeps track of the most dominant language and will return that language.
but if the first file that is opened will contain a SpanDurationBreakdown it may use the NoOpLanguageService and that's a bug.

**Comment by shalom938** [comment]
fixed
